src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT = 4;
+    private static final int FIGURE_PARAMETERS = 5;
+    private static final int DEFAULT_FIGURE_PARAMETERS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        if (getQuantityIndex() == 0) {
+            Square square = new Square(getFigureParameters());
+            square.drawFigure();
+            return square;
```

Comment:
```suggestion
    public Figure getRandomFigure() {
        int randomNumber = getQuantityIndex();
        if (randomNumber == 0) {
            return getRandomSquare();
```
you randomize a number just once,
because in each if the number is different

lets also make methods to get each figure - this would make your code easier to read